### PR TITLE
Add line ending check plugin [Rebase & FF]

### PR DESCRIPTION
**Pull request by makubacki**
## Description

Adds a simple CI plugin to check file line endings. This is meant to:

1. Supplement the Uncrustify Check plugin which fails if line endings
   are CRLF but does not obviously indicate that error since it is
   non-visible whitespace.

2. Discover line endings issues in files not checked by Uncrustify
   (which only operates against .c and .h files right now).

Pre-existing line ending issues (LF instead of CRLF are fixed in the first commit).

- [ ] Breaking change?
  - Will this change break pre-existing builds or functionality without action being taken?
  **Possibly** - This plugin will break the CI build for any package that does not have CRLF
  line endings.

## How This Was Tested

1. Run against all packages in mu_basecore and other platform packages.
2. Verified ignore file list works as expected.
3. Verified line endings are interpreted as expected.
4. Verified plugin returns success and failure in expected cases.

## Integration Instructions

Check plugin results when integrating this mu_basecore change.

### CODE REVIEW

File: .pytool/Plugin/LineEndingCheck/LineEndingCheck.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+        if abs_pkg_path is None:
+            tc.SetSkipped()
+            tc.LogStdError(f"Package folder not found {abs_pkg_path}")
+            return 0
+
+        all_files = [n for n in glob.glob(os.path.join(abs_pkg_path, '*.*'),
+                     recursive=True)]
+
+        if "IgnoreFiles" in package_config:
```
Comment (spbrogan) [inline .pytool/Plugin/LineEndingCheck/LineEndingCheck.py]: why not leverage the more capable gitignore syntax?

File: .pytool/Plugin/LineEndingCheck/Readme.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Line Ending Check Plugin
+
+This CiBuildPlugin scans all the files in a package to verify that the line endings are CRLF.
+
+## Configuration
+
+The plugin can be configured to ignore certain files.
+
+``` yaml
+"CharEncodingCheck": {
```
Comment (spbrogan) [inline .pytool/Plugin/LineEndingCheck/Readme.md]: this looks incorrect.  Copy and paste error?

File: .pytool/Plugin/LineEndingCheck/LineEndingCheck.py (hunk 1 above)
Comment (makubacki) [reply .pytool/Plugin/LineEndingCheck/LineEndingCheck.py]: Updated to use gitignore syntax

File: .pytool/Plugin/LineEndingCheck/Readme.md (hunk 2 above)
Comment (makubacki) [reply .pytool/Plugin/LineEndingCheck/Readme.md]: Yes. Forgot to search the readme. Updated.

**Comment by makubacki** [comment]
@spbrogan, all feedback is addressed.

File: .pytool/Plugin/LineEndingCheck/Readme.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+"LineEndingCheck": {
+    "IgnoreFiles": []
+}
+```
+
+### IgnoreFiles
+
+An **optional** list of Unix style filename patterns used to exclude files from being checked.
+
+See [fnmatch](https://docs.python.org/3/library/fnmatch.html#module-fnmatch) for more details.
```
Comment (spbrogan) [inline .pytool/Plugin/LineEndingCheck/Readme.md]: is fnmatch still relevant?

File: .pytool/Plugin/LineEndingCheck/Readme.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+``` yaml
+"LineEndingCheck": {
+    "IgnoreFiles": []
+}
+```
+
+### IgnoreFiles
+
+An **optional** list of Unix style filename patterns used to exclude files from being checked.
```
Comment (spbrogan) [inline .pytool/Plugin/LineEndingCheck/Readme.md]: Is this still accurate?  Are we saying that .gitignore syntax is the same as unix style filename pattern?  Doesn't it also work with directories?  Should we mention, they must be relative from package root?

File: .pytool/Plugin/LineEndingCheck/Readme.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# Line Ending Check Plugin
```
Comment (spbrogan) [inline .pytool/Plugin/LineEndingCheck/Readme.md:1]: Should there also be a comment that this plugin is really only relevant if the workspace is checked out on Linux or with autocrlf=false

File: .pytool/Plugin/LineEndingCheck/Readme.md (hunk 3 above)
Comment (makubacki) [reply .pytool/Plugin/LineEndingCheck/Readme.md]: No. I removed this section.

File: .pytool/Plugin/LineEndingCheck/Readme.md (hunk 5 above)
Comment (makubacki) [reply .pytool/Plugin/LineEndingCheck/Readme.md:1]: Added guidance in https://github.com/microsoft/mu_basecore/pull/123/commits/d69740acdb0747e1e37dd1c867a271212e16b73c

File: .pytool/Plugin/LineEndingCheck/Readme.md (hunk 4 above)
Comment (makubacki) [reply .pytool/Plugin/LineEndingCheck/Readme.md]: Yes, the git ignore syntax is more robust. Along with the fnmatch link, I forgot to update this section. It's updated now.

https://github.com/microsoft/mu_basecore/pull/123/commits/7211bef603a41285dd60cb6de852f7cfb1b04e8c

File: .pytool/Plugin/LineEndingCheck/Readme.md (hunk 3 above)
Comment (makubacki) [reply .pytool/Plugin/LineEndingCheck/Readme.md]: https://github.com/microsoft/mu_basecore/pull/123/commits/7211bef603a41285dd60cb6de852f7cfb1b04e8c

**Review by spbrogan — APPROVED** [review]

**Pull request merged**